firstvoices/backend/serializers/fields.py

```diff
@@ -27,7 +27,11 @@ def get_url(self, obj, view_name, request, format):
         kwargs = {self.lookup_url_kwarg: lookup_value}
 
         # add the site lookup
-        kwargs.update({"site_slug": get_site_from_context(self).slug})
+        try:
+            kwargs.update({"site_slug": get_site_from_context(self).slug})
+        except AttributeError:
+            # If site not present
+            kwargs.update({"site_slug": None})
```

Comment:
Why is this necessary? This field is only meant to be used for site content urls that always have a site slug

Reply:
Yes, So It seems that the error is arising due to the usage of the `DictionaryDetailSerializer`. This serializer tries to add the `url` field, which requires the site to be in context in order to form the URL. However, in the case of the search API, which is a top-level API, the site is not included in the URL but passed as a parameter. As a result, an exception is thrown.
To address this problem, I implemented a workaround in the `utils.py `file. The conditional clause there checks if the site is present in the context and returns None if it is not. However, accessing this `None` value may result in an AttributeError, So I also added a try/catch block to handle that, as per Sonar's suggestion.

Reply:
If this is called with no site_slug available, it will fail to correctly reverse the url though, right? So it seems strange to prevent the error when it is an actual problem

Reply:
I think we need to solve this a different way, and intentionally make the DictionaryEntrySerializer work outside of a site content api.